src/lib/y2storage/package_handler.rb

```diff
@@ -73,7 +73,22 @@ def add_packages(pkg_list)
     #
     def add_feature_packages(devicegraph)
       used_features = UsedStorageFeatures.new(devicegraph)
-      add_packages(used_features.feature_packages)
+      packages = used_features.feature_packages
+      packages = packages.delete_if { |pkg| unavailable_optional_package?(pkg) }
+      add_packages(packages)
+    end
+
+    # Check if a package is an optional package that is unavailable.
+    # See also bsc#1039830
+    #
+    # @param package [String] package name
+    # @return [Boolean] true if package is optional and unavailable,
+    #                   false if not optional or if available.
+    def unavailable_optional_package?(package)
+      return false unless UsedStorageFeatures::OPTIONAL_PACKAGES.include?(package)
+      return false if Yast::Package.Available(package)
+      log.warn("WARNING: Skipping unavailable filesystem support package #{package}")
```

Comment:
NP: IMO it should be a responsibility of the caller to log the fact that this package is being removed from the list (skipped). This method should only return `true` or `false` and make no assumptions about the consequences.

In other words, the code deleting something from the list and the code logging that something was deleted should be in the same place. This method is actually not skipping the package, the `delete` is doing that.